### PR TITLE
base image for OpenShift CI integration

### PR DESCRIPTION
**Pull request by andfasano**
Add base image that will be used to run the e2e-metal-ipi test

### CODE REVIEW

**Comment by stbenjam** [comment]
This should contain the dev-scripts contents and the /etc/passwd permissions changes we use elsewhere 

https://github.com/openshift/installer/blob/6decc16fbd34f13adc698914fc4be09de6a896f7/images/baremetal/Dockerfile.ci#L8

**Comment by stbenjam** [comment]
Also should probably be based on ubi8 not centos

**Comment by andfasano** [comment]
As a first step I wanted just to address the packet setup/teardown step, and add the rest later. Any specific reason to stick to ubi8 and not to a general purpose image, considering that's used only for testing?

**Comment by stbenjam** [comment]
There's not any particular reason for ubi8 over centos8 than consistency. You could also use one of the OCP base images: registry.svc.ci.openshift.org/origin/4.5:base.  I just don't know of anything else using CentOS.

File: images/Dockerfile.ci [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# This Dockerfile builds the image used by the e2e-metal-ipi test steps in the OpenShift CI.
+# For more details about the test see https://steps.svc.ci.openshift.org/job/openshift-baremetal-operator-master-e2e-metal-ipi
+
+FROM centos:8
+RUN dnf install epel-release -y epel-release && \
+    dnf install -y ansible
+RUN pip3 install packet-python
```
Comment (elfosardo) [inline images/Dockerfile.ci:7]: maybe just continue the precedent RUN so we save one layer
also add a dnf clean all at the end, will help reducing the size of the image

File: images/Dockerfile.ci [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# This Dockerfile builds the image used by the e2e-metal-ipi test steps in the OpenShift CI.
+# For more details about the test see https://steps.svc.ci.openshift.org/job/openshift-baremetal-operator-master-e2e-metal-ipi
+
+FROM centos:8
+RUN dnf install epel-release -y epel-release && \
```
Comment (elfosardo) [inline images/Dockerfile.ci:5]: you put epel-release twice :)

Comment (andfasano) [reply images/Dockerfile.ci:5]: Oops right!

**Comment by elfosardo** [comment]
> There's not any particular reason for ubi8 over centos8 than consistency. You could also use one of the OCP base images: registry.svc.ci.openshift.org/origin/4.5:base. I just don't know of anything else using CentOS.

please consider that no images in openshift at the moment use anything ending with an 8, they're all rhel7 based AFAIK

**Comment by andfasano** [comment]
We could use then the `tests` image as base: https://github.com/openshift/origin/blob/master/images/tests/Dockerfile.rhel. It's already been used by one of the step, it could become then the only image used for the workflow.

File: images/Dockerfile.ci [hunk 3]
```diff
@@ -1,7 +1,10 @@
 # This Dockerfile builds the image used by the e2e-metal-ipi test steps in the OpenShift CI.
 # For more details about the test see https://steps.svc.ci.openshift.org/job/openshift-baremetal-operator-master-e2e-metal-ipi
 
-FROM centos:8
-RUN dnf install epel-release -y epel-release && \
-    dnf install -y ansible
-RUN pip3 install packet-python
+FROM openshift/ubi:8
+RUN yum -y update && \
```
Comment (elfosardo) [inline images/Dockerfile.ci:5]: use dnf, yum is deprecated :)
also no need to run update/upgrade, it's a general anti-pattern in containers (I know we do it in other images, but we really shouldn't.... )

**Pull request merged**